src/App.tsx

```diff
@@ -26,70 +27,135 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((prevGood, currGood) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return prevGood.localeCompare(currGood);
+
+      case SortType.LENGTH:
+        return prevGood.length - currGood.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortByType = (type: SortType) => {
+    this.setState({ sortType: type });
+  };
+
+  reset = () => {
+    this.sortByType(SortType.NONE);
+    this.setState({ isReversed: false });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const showResetButton = sortType !== SortType.NONE || isReversed;
+    const reorderedGoods = getReorderedGoods(
+      goodsFromServer, this.state,
+    );
```

Comment:
No need to wrap arguments here

```suggestion
    const reorderedGoods = getReorderedGoods(goodsFromServer, this.state);
```